groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf

```diff
@@ -0,0 +1,175 @@
+# ------------------------------------------------------------------------------
+# ABBYY Document Capture & OCR Security Group and rules
+# ------------------------------------------------------------------------------
+module "abbyy_doc_ocr_ec2_security_group" {
+  source  = "terraform-aws-modules/security-group/aws"
+  version = "~> 3.0"
+
+  name        = "sgr-${var.application}-abbyy-doc-ocr-server"
+  description = "Security group for the ${var.application} ABBYY Document Capture & OCR Server"
+  vpc_id      = data.aws_vpc.vpc.id
+
+  ingress_with_cidr_blocks = [
+    {
+      from_port   = 135
+      to_port     = 135
+      protocol    = "tcp"
+      cidr_blocks = join(",", local.abbyy_doc_ocr_135_cidr_block)
+    },
+    {
+      from_port   = 445
+      to_port     = 445
+      protocol    = "tcp"
+      cidr_blocks = "172.16.101.82/32"
+    },
+    {
+      from_port   = 1000
+      to_port     = 1000
+      protocol    = "tcp"
+      cidr_blocks = "172.16.200.58/32"
+    },
+    {
+      rule        = "rdp-tcp"
+      cidr_blocks = join(",", local.abbyy_doc_ocr_rdp_cidr_block)
+    },
+    {
+      rule        = "winrm-http-tcp"
+      cidr_blocks = "172.16.155.9/32"
+    },
+    {
+      from_port   = 6129
+      to_port     = 6129
+      protocol    = "tcp"
+      cidr_blocks = "10.172.0.0/17"
+    },
+    {
+      from_port   = 49155
+      to_port     = 49155
+      protocol    = "tcp"
+      cidr_blocks = join(",", local.abbyy_doc_ocr_49155_cidr_block)
+    },
+    {
+      from_port   = 53853
+      to_port     = 53853
+      protocol    = "tcp"
+      cidr_blocks = "172.16.101.82/32"
+    },
+    {
+      from_port   = 49227
+      to_port     = 49227
+      protocol    = "tcp"
+      cidr_blocks = "172.16.101.82/32"
+    },
+    #{
+    #rule        = "rdp-tcp"
+    #cidr_blocks = join(",", local.abbyy_doc_ocr_third_party_rdp_cidr_block)
+    #}
+  ]
+
+  computed_ingress_with_source_security_group_id = [
+    {
+      rule                     = "http-80-tcp"
+      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
+    },
+    {
+      from_port                = 10020
+      to_port                  = 10020
+      protocol                 = "tcp"
+      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
+    },
```

Comment:
You can probably replace these with a "self" rule that the module provides, should cover all possible ports between servers in the same security group.

Reply:
updated as follows:

ingress_with_self = [
    {
      rule                     = "http-80-tcp"
      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
    },
    {
      from_port                = 10020
      to_port                  = 10020
      protocol                 = "tcp"
      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
    },
    {
      rule                     = "http-80-tcp"
      source_security_group_id = module.abbyy_doc_ocr_internal_alb_security_group.this_security_group_id
    }
  ]